tests/integration/test_group_role_mapping.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import os
+from google.auth import default
+from google.auth.transport.requests import Request
+import sqlalchemy
+from aiohttp import ClientSession
+from helpers import delete_database_user
+from iam_groups_authn.mysql import init_mysql_connection_engine, mysql_username
+from iam_groups_authn.sql_admin import get_instance_users
+from iam_groups_authn.sync import GroupRoleMaxLengthError, groups_sync, UserService
+
+# load test params from environment
+sql_instance = os.environ["MYSQL_INSTANCE"]
+long_iam_group = [os.environ["LONG_GROUP_EMAIL"]]
+test_user = os.environ["TEST_USER"]
+
+scopes = [
+    "https://www.googleapis.com/auth/admin.directory.group.member",
+    "https://www.googleapis.com/auth/sqlservice.admin",
+]
+
+
+def check_role_mysql(db, role):
+    """Function to get database users who have given MySQL role.
+
+    Args:
+        db: Database connection pool to connect to.
+        role: Given role to query users in database with.
+
+    Returns:
+        users_with_role: List of users who have role granted to them.
+    """
+
+    stmt = sqlalchemy.text(
+        "SELECT TO_USER FROM mysql.role_edges WHERE FROM_USER= :role"
+    )
+    with db.connect() as conn:
+        results = conn.execute(stmt, {"role": role}).fetchall()
+    users_with_role = [result[0] for result in results]
+    return users_with_role
+
+
+@pytest.fixture(name="credentials", autouse=True)
+def setup_and_teardown():
+    """Function for setting up and tearing down test."""
+
+    # load in service account credentials for test
+    credentials, _ = default(scopes=scopes)
+
+    # check if credentials are expired
+    if not credentials.valid:
+        request = Request()
+        credentials.refresh(request)
+
+    yield credentials
+
+    try:
+        # cleanup group role from database
+        delete_database_user(sql_instance, "short-group-role", credentials)
+    except Exception:
+        print("------------------------Cleanup Failed!------------------------")
```

Comment:
should this raise the exception as well, so the test fails?

Reply:
This cleanup is used for both tests. The one test fails to add a user so the cleanup will fail since the user does not exist. The cleanup is fine failing for this case.